src/Nuget.PackageIndex.VisualStudio/Analyzers/ProjectTargetFrameworkProviderExporter.cs

```diff
@@ -104,56 +105,56 @@ public IEnumerable<TargetFrameworkMetadata> GetTargetFrameworks(string filePath)
 
-                    var provider = Providers.FirstOrDefault(x => x.SupportsProject(dteProject));
+                    var provider = Providers.FirstOrDefault(x => x.SupportsProject(dteProject.FullName));
                     if (provider != null)
                     {
-                        uniqueProjectName = dteProject.UniqueName;
+                        projectFullPath = dteProject.FullName;
                         resultFrameworks = provider.GetTargetFrameworks(dteProject);
                     }
                 }
                 catch (Exception e)
                 {
                     // Add to Package Manager console?
-                    Debug.Write(string.Format("{0}. Stack trace: {1}", e.Message, e.StackTrace));
+                    Debug.Write(e.ToString());
                 }
             });
 
             // add file and project frameworks to cache
             lock(_cacheLock)
             {
-                if (ProjectFrameworksCache.Keys.Contains(uniqueProjectName))
+                if (ProjectFrameworksCache.Keys.Contains(projectFullPath))
                 {
-                    ProjectFrameworksCache[uniqueProjectName] = resultFrameworks;
+                    ProjectFrameworksCache[projectFullPath] = resultFrameworks;
                 }
                 else
                 {
-                    ProjectFrameworksCache.Add(uniqueProjectName, resultFrameworks);
+                    ProjectFrameworksCache.Add(projectFullPath, resultFrameworks);
                 }
 
                 if (FilesCache.Keys.Contains(filePath))
                 {
-                    FilesCache[filePath] = uniqueProjectName;
+                    FilesCache[filePath] = projectFullPath;
                 }
                 else
                 {
-                    FilesCache.Add(filePath, uniqueProjectName);
+                    FilesCache.Add(filePath, projectFullPath);
                 }
             }
 
             return resultFrameworks;
         }
 
-        private void OnProjectTargetFrameworkChanged(object sender, EnvDTE.Project project)
+        private void OnProjectTargetFrameworkChanged(object sender, string projectFullPath)
         {
             lock (_cacheLock)
             {
                 // here we need to remove from cache files that belong to a given project
-                if (ProjectFrameworksCache.Keys.Contains(project.UniqueName))
+                if (ProjectFrameworksCache.Keys.Contains(projectFullPath))
                 {
-                    ProjectFrameworksCache.Remove(project.UniqueName);
+                    ProjectFrameworksCache.Remove(projectFullPath);
                 }
 
-                var filesToRemove = FilesCache.Where(x => x.Value.Equals(project.UniqueName, StringComparison.InvariantCultureIgnoreCase)).Select(x => x.Key).ToList();
+                var filesToRemove = FilesCache.Where(x => x.Value.Equals(projectFullPath, StringComparison.OrdinalIgnoreCase)).Select(x => x.Key).ToList();
```

Comment:
Another small note - Because this code goes on github, its always a good idea to break long line. A good rule of thumb is that if you can't see the end of a line in the PR you want to break it.